ceph_deploy/osd.py

```diff
@@ -388,6 +388,228 @@ def activate(args, cfg):
         distro.conn.exit()
 
 
+def destroy(args):
+    cluster = args.cluster
+
+    for hostname, disk, journal in args.disk:
+        if not hostname:
+            raise RuntimeError('invalided hostname')
+        LOG.debug(
+            'Destroy osd id %s in host %s',
+            args.osd_id, hostname
+        )
+
+        distro = hosts.get(hostname, username=args.username)
+        LOG.info(
+            'Distro info: %s %s %s',
+            distro.name,
+            distro.release,
+            distro.codename
+        )
+
+        destroy_osd(distro, cluster, args.osd_id)
+
+
+def destroy_osd(distro, cluster, osd_id):
+
+    found_in_node = False
+    found_in_stray = False
+    cluster_path = paths.osd.base(cluster)
+    conn = distro.conn
+
+    if not osd_id or \
+       osd_id is None:
+        LOG.info("(NOT IMPLEMENT) remove all osd")
+        raise NotImplementedError("It need to assign the specific osd id.")
+
+    command = [
+        'ceph',
+        '--cluster={cluster}'.format(cluster=cluster),
+        'osd',
+        'tree',
+        '--format=json',
+    ]
```

Comment:
I completely agree your opinion.
It sounds more sensible abut we use the osd_tree() function.

OK, I think I will check osd_list() and osd_tree() first.
Then the better way is that we can gather the osd tree once from **monitor** node when we call destroy() function by osd_list()/osd_tree() ?